### PR TITLE
Fix minor typos in docstrings

### LINKED ISSUE #884: [DOC] Fix docstrings for doc sprint
#### Describe the issue linked to the documentation
#863 

#### Easy files to fix
* https://github.com/alan-turing-institute/sktime/blob/77c2547b7f05e2975962eca749d4c9f4f2b8468e/sktime/utils/_maint/_show_versions.py
* https://github.com/alan-turing-institute/sktime/blob/main/sktime/transformations/series/boxcox.py
* https://github.com/alan-turing-institute/sktime/blob/main/sktime/transformations/series/acf.py

### PR DESCRIPTION
**Pull request by GuzalBulatova**
#### What does this implement/fix? Explain your changes.
Fixes minor typos in `transformations/series/boxcox.py`

#### Does your contribution introduce a new dependency? If yes, which one?

No.

#### What should a reviewer concentrate their feedback on?

The function `_eval_pearsonr` is an inline function, I've slightly changed the docstring, but haven't converted it to a standard function docstr. Is it ok?

### CODE REVIEW

**Comment by GuzalBulatova** [comment]
Pydocstyle's suggestions seem irrelevant in these cases to me.

1. Should I add docstring to these 3 methods?

    `line 47 in public method `fit`:
        D102: Missing docstring in public method`

    `line 57 in public method `transform`:
        D102: Missing docstring in public method`

    `line 63 in public method `inverse_transform`:
        D102: Missing docstring in public method`

2. And I'm still not sure about `_eval_pearsonr`:

    `sktime/transformations/series/boxcox.py:114 in private nested function `_eval_pearsonr`:
        D205: 1 blank line required between summary line and description (found 0)`

    `sktime/transformations/series/boxcox.py:114 in private nested function `_eval_pearsonr`:
        D400: First line should end with a period (not 't')`

**Comment by mloning** [comment]
Hi @GuzalBulatova, 

1. Yes, please add a docstring for those methods, I fixed the docstrings in this file: https://github.com/alan-turing-institute/sktime/blob/main/sktime/transformations/series/acf.py - that should help you fill in the docstrings for `fit`, `transform` and `inverse_transform`
2. Seems like you can either remove the docstring for private functions entirely or fix it (see http://www.pydocstyle.org/en/stable/error_codes.html#publicity)
3. Thanks for raising these questions, I feel that fixing the documentation is very important for us at this stage and that we can still achieve this through the automated CI checks but happy to discuss this again in a few weeks if it causes too much trouble!

**Comment by GuzalBulatova** [comment]
Hi @mloning ! I've added docstrings to fit, transform and inverse_transform , thanks for the hint, it helped a lot.
I looked at the pydocstyle link you sent, thank you, it's a great resource! Removed docstr for _eval_pearsonr as it's not required for inside nested function.
I think now the PR is ready for review.

File: sktime/transformations/series/boxcox.py [hunk 1]
```diff
@@ -45,6 +45,20 @@ def __init__(self, bounds=None, method="mle", sp=None):
...
+        Parameters
+        ----------
+        Z : pd.Series
+            Series to fit.
+        X : pd.DataFrame, optional (default=None)
+            Exogenous data used in transformation.
+
+        Returns
+        -------
+        Zt : pd.Series
```
Comment (mloning) [inline sktime/transformations/series/boxcox.py:59]: Fit always returns `self`

Comment (GuzalBulatova) [reply sktime/transformations/series/boxcox.py]: Thanks, I didn't know this! Have changed this line.

**Review by mloning — APPROVED** [review]
Looks great - will merge this now @GuzalBulatova!

**Pull request merged**